src/import_export_job/tasks.py

```diff
@@ -37,61 +37,45 @@ def get_format(job):
 
 
 def _run_import_job(import_job, dry_run=True):
-    change_job_status(import_job, "import", "1/5 Import started", dry_run)
+    def update_status(step, message):
+        change_job_status(import_job, "import", f"{step} {message}", dry_run)
+
+    def log_error(message, exception=None):
+        error_message = _(message)
+        if exception:
+            error_message += f": {exception}"
+        import_job.errors += error_message + "\n"
+        update_status("Error", error_message)
+        import_job.save()
+
+    update_status("1/4", "Import started")
     import_job.errors = ""
 
     model_config = ModelConfig(**importables[import_job.model])
-
     import_format = get_format(import_job)
 
-    # Copied from https://github.com/django-import-export/django-import-export/blob/3c082f98afe7996e79f936418fced3094f141c26/import_export/admin.py#L260 sorry  # noqa
     try:
-        data = import_job.file.read()
-
-        if isinstance(data, bytes):
-            data = data.decode("utf8")
+        data_chunks = []
+        with import_job.file.open('rb') as file:  # Open in binary mode
+            for chunk in file.chunks():
+                data_chunks.append(chunk.decode('utf-8'))
+        data = ''.join(data_chunks)
```

Comment:
What is the benefit of:
1. Reading this file in binary mode, and then decoding it later, over using file.open('r', encoding='utf-8')?
2. Reading the file into chunks and then joining them together. The variable data still contains the complete file. So what is the benefit of chunking?

Reply:
removed